python/generate.py

```diff
@@ -456,8 +458,10 @@ def get_obj_docstring_dict(obj, key_prefix, signature_prefix, only_include=None)
                 is_property,
             )
 
-        # Add the extracted metadata to obj_docstring_dict
-        obj_docstring_dict[fullname] = member_docstring_dict
+        # Add the extracted metadata to obj_docstring_dict if the object is
+        # local to streamlit (null source occurs when the object is inherited
+        if member_docstring_dict["source"]:
+            obj_docstring_dict[fullname] = member_docstring_dict
```

Comment:
What is `member_docstring_dict["source"]:` doing? From going through the code, it looks like `member_docstring_dict` does not have a source attribute (it was never assigned; it's assigned to `description`).

Wouldn't running `python generate.py` result in:

```sh
Traceback (most recent call last):
  File "/docs/python/generate.py", line 551, in <module>
    data = get_streamlit_docstring_dict()
  File "/docs/python/generate.py", line 533, in get_streamlit_docstring_dict
    module_docstring_dict.update(get_obj_docstring_dict(obj, *key))
  File "/docs/python/generate.py", line 463, in get_obj_docstring_dict
    if member_docstring_dict["source"]:
KeyError: 'source'
```

Reply:
Ah, I see. The script doesn't error out only when run on the https://github.com/streamlit/streamlit/tree/docs/docstring-edits branch.